Autosplitters.Tests/VersionTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿using FluentAssertions;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace Autosplitters.Tests;
+
+public class VersionTests
+{
+    [Theory]
+    [InlineData("TR123", "TR123")]
+    [InlineData("TombRaider1996", "TR1996")]
+    [InlineData("TombRaiderII", "TR2")]
+    [InlineData("TombRaiderIII", "TR3")]
+    [InlineData("TombRaiderIV", "TR4")]
+    [InlineData("TombRaiderV", "TR5")]
+    public void Version_Strings_Should_Be_Equal(string project, string dll)
+    {
+        var currentDirectory = Directory.GetCurrentDirectory(); // autosplitters\Autosplitters.Tests\bin\[Debug/Release]
+
+        var assemblyInfoPath = Path.Combine(currentDirectory, $"../../../{project}/Properties/AssemblyInfo.cs");
+        var assemblyInfoContents = File.ReadAllText(assemblyInfoPath);
+        var assemblyInfoVersionString = Regex.Matches(assemblyInfoContents, @"\[assembly: AssemblyVersion\(""(?<VersionString>.+)""\)]")
+                                             .Cast<Match>()
+                                             .Last()
+                                             .Groups["VersionString"]
+                                             .Value;
+
+        var updateXmlPath = Path.Combine(currentDirectory, $"../../../{project}/Components/update.xml");
+        var updateXmlContents = File.ReadAllText(updateXmlPath);
+        var updateXmlVersionString = Regex.Match(updateXmlContents, @"<update version=""(?<VersionString>.+)"">")
+                                          .Groups["VersionString"]
+                                          .Value;
+
+        var dllPath = Path.Combine(currentDirectory, $"../../../{project}/Components/{dll}.dll");
+        var dllVersionInfo = FileVersionInfo.GetVersionInfo(dllPath).FileVersion;
+
+        assemblyInfoVersionString.Should().Be(dllVersionInfo);
+        updateXmlVersionString.Should().Be(dllVersionInfo);
+    }
```

Comment:
2 notes:
1. Recommend normalizing Revisions and using `Version.Equals` (shown in code suggestion)
2. Optional style consideration: raw strings `"""` instead of `@"` for Regex strings (not shown in code suggestion)

```suggestion
    {
        var currentDirectory = Directory.GetCurrentDirectory(); // autosplitters\Autosplitters.Tests\bin\[Debug/Release]

        var assemblyInfoPath = Path.Combine(currentDirectory, $"../../../{project}/Properties/AssemblyInfo.cs");
        var assemblyInfoContents = File.ReadAllText(assemblyInfoPath);
        var assemblyInfoVersionString = Regex.Matches(assemblyInfoContents, @"\[assembly: AssemblyVersion\(""(?<VersionString>.+)""\)]")
                                             .Cast<Match>()
                                             .Last()
                                             .Groups["VersionString"]
                                             .Value;
        var assemblyInfoVersion = new Version(assemblyInfoVersionString);
        assemblyInfoVersion = NormalizeRevisionIfMissing(assemblyInfoVersion);

        var updateXmlPath = Path.Combine(currentDirectory, $"../../../{project}/Components/update.xml");
        var updateXmlContents = File.ReadAllText(updateXmlPath);
        var updateXmlVersionString = Regex.Match(updateXmlContents, @"<update version=""(?<VersionString>.+)"">")
                                          .Groups["VersionString"]
                                          .Value;
        var updateXmlVersion = new Version(updateXmlVersionString);
        updateXmlVersion = NormalizeRevisionIfMissing(updateXmlVersion);

        var dllPath = Path.Combine(currentDirectory, $"../../../{project}/Components/{dll}.dll");
        var dllVersionInfo = FileVersionInfo.GetVersionInfo(dllPath).FileVersion;
        var dllVersion = new Version(dllVersionInfo);
        dllVersion = NormalizeRevisionIfMissing(dllVersion);

        assemblyInfoVersion.Equals(dllVersion).Should().BeTrue($"Assembly Info version {assemblyInfoVersionString} must match DLL version {dllVersionInfo}");
        updateXmlVersion.Equals(dllVersion).Should().BeTrue($"Update XML version {updateXmlVersionString} must match DLL version {dllVersionInfo}");

        return;

        static Version NormalizeRevisionIfMissing(Version version)
        {
            return version.Revision == -1
                ? new Version(version.Major, version.Minor, version.Build, 0)
                : version;
        }
    }
```